From 068e1cc6d70b2266605527c4e00b7fc09afc2418 Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Tue, 30 Aug 2022 15:45:10 +0200
Subject: [PATCH 17/52] Segmentation fault of tools/registry during compilation

TYPE: Debug

KEYWORDS: tools/registry, Registry

Modified Files:
  tools/gen_interp.c

Description: When many variables are added in Registry, tools/registry segfaults.
While tools/registry goes through Registry files,
it creates a temporary string in C function contains_tok by adding all registry variables separated by commas.
You can find the function in tools/gen_interp.c.
The default number of string length allocated to tempstr is 8092.
For this commit, 8092 is increased to 9092.

SOURCE: Tatsuo ONISHI
---
 tools/gen_interp.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/tools/gen_interp.c b/tools/gen_interp.c
index e2a49ae9..23db3e11 100644
--- a/tools/gen_interp.c
+++ b/tools/gen_interp.c
@@ -52,8 +52,10 @@ int contains_str( char *s1, char *s2 )
 int contains_tok( char *s1, char *s2, char *delims )
 {
   char *p ;
-  char tempstr[8092] ;
-
+  /* TO2 */
+  /* char tempstr[8092] ; */
+  char tempstr[9092] ;
+  /* TO2 */
   strcpy( tempstr , s1 ) ;
   p = strtok ( tempstr, delims ) ;
   while ( p != NULL )
-- 
2.31.1

